api/machine_types.go

```diff
@@ -354,11 +354,12 @@ func (mp *MachinePort) HasNonHttpPorts() bool {
 }
 
 type MachineService struct {
-	Protocol     string                     `json:"protocol,omitempty" toml:"protocol,omitempty"`
-	InternalPort int                        `json:"internal_port,omitempty" toml:"internal_port,omitempty"`
-	Ports        []MachinePort              `json:"ports,omitempty" toml:"ports,omitempty"`
-	Checks       []MachineCheck             `json:"checks,omitempty" toml:"checks,omitempty"`
-	Concurrency  *MachineServiceConcurrency `json:"concurrency,omitempty" toml:"concurrency"`
+	Protocol          string                     `json:"protocol,omitempty" toml:"protocol,omitempty"`
+	InternalPort      int                        `json:"internal_port,omitempty" toml:"internal_port,omitempty"`
+	Ports             []MachinePort              `json:"ports,omitempty" toml:"ports,omitempty"`
+	Checks            []MachineCheck             `json:"checks,omitempty" toml:"checks,omitempty"`
+	Concurrency       *MachineServiceConcurrency `json:"concurrency,omitempty" toml:"concurrency"`
+	AutostartMachines bool                       `json:"autostart_machines,omitempty" toml:"autostart_machines,omitempty"`
```

Comment:
should the default be off or on? I think the default is "on" today (before this change), and I imagine we want to preserve that.

If so, how about making this `NoAutostartMachines`? that way the default when unset will be `false` (golang's default for bool type).

```suggestion
	NoAutostartMachines bool                       `json:"no_autostart_machines,omitempty" toml:"no_autostart_machines,omitempty"`
```

Reply:
Updating to work in line with changes in `web` (see comment here that's inline with your thinking https://github.com/superfly/web/pull/1439#issuecomment-1428315679)